src/utils/libfabric/libfabric_common.cpp

```
@@ -178,19 +178,9 @@ uint16_t
-        while (expected > NIXL_XFER_ID_MASK &&
-               !g_xfer_id_counter.compare_exchange_weak(expected, 1)) {
-            expected = g_xfer_id_counter.load();
-        }
+    // uint16_t wraps naturally from 65535 back to 0; skip 0 since it is used as a sentinel value
+    if (xfer_id == 0) {
         xfer_id = g_xfer_id_counter.fetch_add(1);
-        // Ensure we don't exceed the mask after reset
-        if (xfer_id > NIXL_XFER_ID_MASK) {
-            xfer_id = 1;
-        }
     }
```

Cppcheck (2.20.0)

[style] The function 'setPayload' is never used.

(unusedFunction)